### PR TITLE
Add Threefry PRNG and tests

### PR DESCRIPTION
**Pull request by austinzheng**
This PR adds a PRNG (conforming to S4TF's `SeedableRandomNumberGenerator` protocol), as well as tests that ensure that it is not accidentally broken by future changes.

Please let me know if this directory structure/naming is appropriate, or if there is any refactoring you wish for me to do.

ref: https://github.com/apple/swift/pull/22318

### CODE REVIEW

File: Sources/DeepLearning/Random.swift [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        X1 = rotl32(value: X1, n: rot.3)
+        X1 ^= X0
+        // Key injection (r = 5)
+        X0 &+= ks2
+        X1 &+= (ks0 + 5)
+
+        return (X0, X1)
+    }
+
+    init(uint64seed seed: UInt64) {
```
Comment (rxwei) [inline Sources/DeepLearning/Random.swift]:
```suggestion
    private init(uint64Seed seed: UInt64) {
```

**Comment by austinzheng** [comment]
Hmm...I applied the suggested change using the GitHub interface, and now the CLA bot is complaining. Since we're both members of the `Google` organization, it shouldn't be doing that.

**Comment by rxwei** [comment]
Nvm the bot :)

**Comment by austinzheng** [comment]
Bot is blocking me from merging. Is there a way to override? (Sorry, next time I'll just make the suggested changes manually and push another commit.)

File: Sources/DeepLearning/Random.swift [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        precondition(seed.count <= 8, "Length of seed must be at most 8")
+        var combinedSeed: UInt64 = 0
+        for i in 0..<seed.count {
+            combinedSeed += UInt64(seed[i]) << UInt64(8 * i)
+        }
+        self.init(uint64seed: combinedSeed)
+    }
+
+    public mutating func next() -> UInt64 {
+        let value = ThreefryRandomNumberGenerator.combine(
```
Comment (rxwei) [inline Sources/DeepLearning/Random.swift]: This may look slightly nicer if you do
```swift
defer { ctr += 1 }
return Threefry...
```

**Comment by rxwei** [comment]
I think i can override this.

File: Sources/DeepLearning/Random.swift (hunk 2 above)
Comment (austinzheng) [reply Sources/DeepLearning/Random.swift]: Done!

**Comment by austinzheng** [comment]
Through the magic of git history rewriting, googlebot no longer spuriously complains!

**Comment by austinzheng** [comment]
Now that I know how all the tooling works, I should have CLs for Philox and a native ops PRNG soon, hopefully.

**Pull request merged**